go/tasks/v1/flytek8s/plugin_executor.go

```diff
@@ -243,6 +244,38 @@ func (e *K8sTaskExecutor) CheckTaskStatus(ctx context.Context, taskCtx types.Tas
 		PhaseVersion: taskCtx.GetPhaseVersion(),
 	}
 
+	// NOTE: To ensure objects are cleaned up, the plugins need a persistent step in addition to upstream plugin executor
+	// state machine. Once the object reaches its terminal state, we commit the completion in two steps:
+	// Round1: mark the object as deleted in state store (object's custom state)
+	// Round2: instead of regular retrieval (which may fail in this case), just delete the object
+	objStatus, terminalPhase, err := retrieveK8sObjectState(taskCtx.GetCustomState())
+	if err != nil {
+		logger.Warningf(ctx, "Failed to retrieve object status: %v. Error: %v",
+			taskCtx.GetTaskExecutionID().GetGeneratedName(), err)
+		return types.TaskStatusUndefined, err
+	}
+	if objStatus == k8sObjectDeleted {
+		// kill the object execution if still live
+		if e.handler.GetProperties().DeleteResourceOnAbort {
+			err = instance.kubeClient.Delete(ctx, o)
```

Comment:
I think we need to call AddObjectMetadata here before calling delete

Reply:
done